dev_guide/builds.adoc

```diff
@@ -1557,3 +1557,114 @@ Docker and Source builds set the following labels on output images:
 |*io.openshift.build.source-location*
 |Source URL for the build
 |===
+
+
+[[using-external-artifacts]]
+= Using External Artifacts During The Build
+
+In order to use external artifacts during the build you don't need any special
+support from OpenShift platform. You could download and use any artifacts
+as you want.
+
+For `*Source*` build strategy you need to put appropriate shell commands into
+the `assemble` script.
+
+.`.sti/bin/assemble` file
+====
+
+[source,bash]
+----
+#!/bin/sh -ex
+GOSU_VERSION=1.7
+wget https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-amd64 -O gosu
+chmod +x gosu
+
+APP_VERSION=1.0
+mkdir /opt/app
+wget -P /opt/app -O app.jar http://repository.example.com/app/app-$APP_VERSION.jar
+----
+====
+
+For `*Docker*` build strategy you need to modify `Dockerfile` and invoke shell
+commands with https://docs.docker.com/engine/reference/builder/#run[`RUN` instruction].
+
+.Excerpt of `Dockerfile`
+====
+
+[source]
+----
+FROM java:8
```

Comment:
Can we use a Java image we provide? `java:8` uses ROOT users, among other pitfalls we already handle in our images.
This is our Java builder image: `fabric8io/java-main`, https://github.com/fabric8io/java-main/tree/master/

But we might want to use another image that is not a builder (need to find the name/link).